sound/soc/intel/boards/sof_realtek_common.c

```diff
@@ -118,28 +172,98 @@ static const struct snd_soc_ops rt1011_ops = {
 static int rt1011_init(struct snd_soc_pcm_runtime *rtd)
 {
 	struct snd_soc_card *card = rtd->card;
+	unsigned int num_codecs = get_num_codecs(RT1011_ACPI_HID);
 	int ret;
 
-	ret = snd_soc_dapm_add_routes(&card->dapm, speaker_map_lr,
-				      ARRAY_SIZE(speaker_map_lr));
-	if (ret)
-		dev_err(rtd->dev, "Speaker map addition failed: %d\n", ret);
+	switch (num_codecs) {
+	case 2:
+		if (!soc_intel_is_cml()) {
+			ret = snd_soc_dapm_add_routes(&card->dapm, speaker_map_lr,
+						      ARRAY_SIZE(speaker_map_lr));
+			if (ret)
+				dev_err(rtd->dev, "fail to add rt1011 routes, ret %d\n",
+					ret);
+			break;
+		}
+
+		/* backward-compatible with existing cml devices */
+		fallthrough;
```

Comment:
when there are 2 codecs we still deal with 4 speakers. is this intended?
how do we differentiate from the case with 4 codecs?